client/asset/btc/spv.go

```diff
@@ -0,0 +1,1712 @@
+	for _, pt := range outpoints {
+		pts = append(pts, &RPCOutpoint{
+			TxID: pt.Txid,
+			Vout: pt.Vout,
+		})
+	}
+	return pts, nil
+}
+
+// changeAddress gets a new internal address from the wallet. The address will
+// be bech32-encoded (P2WPKH).
+func (w *spvWallet) changeAddress() (btcutil.Address, error) {
+	return w.wallet.NewChangeAddress(w.acctNum, waddrmgr.KeyScopeBIP0084)
+}
+
+// AddressPKH gets a new base58-encoded (P2PKH) external address from the
+// wallet.
+func (w *spvWallet) addressPKH() (btcutil.Address, error) {
+	return nil, errors.New("unimplemented")
+}
+
+// addressWPKH gets a new bech32-encoded (P2WPKH) external address from the
+// wallet.
+func (w *spvWallet) addressWPKH() (btcutil.Address, error) {
+	return w.wallet.NewAddress(w.acctNum, waddrmgr.KeyScopeBIP0084)
+}
+
+// signTx attempts to have the wallet sign the transaction inputs.
+func (w *spvWallet) signTx(tx *wire.MsgTx) (*wire.MsgTx, error) {
+	// Can't use btcwallet.Wallet.SignTransaction, because it doesn't work for
+	// segwit transactions (for real?).
+	return tx, w.wallet.signTransaction(tx)
+}
+
+// privKeyForAddress retrieves the private key associated with the specified
+// address.
+func (w *spvWallet) privKeyForAddress(addr string) (*btcec.PrivateKey, error) {
+	a, err := btcutil.DecodeAddress(addr, w.chainParams)
+	if err != nil {
+		return nil, err
+	}
+	return w.wallet.PrivKeyForAddress(a)
+}
+
+// Unlock unlocks the wallet.
+func (w *spvWallet) Unlock(pw []byte) error {
+	return w.wallet.Unlock(pw, nil)
+}
+
+// Lock locks the wallet.
+func (w *spvWallet) Lock() error {
+	w.wallet.Lock()
+	return nil
+}
+
+// sendToAddress sends the amount to the address. feeRate is in units of
+// sats/byte.
+func (w *spvWallet) sendToAddress(address string, value, feeRate uint64, subtract bool) (*chainhash.Hash, error) {
+	addr, err := btcutil.DecodeAddress(address, w.chainParams)
+	if err != nil {
+		return nil, err
+	}
+
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if subtract {
+		return w.sendWithSubtract(pkScript, value, feeRate)
+	}
+
+	wireOP := wire.NewTxOut(int64(value), pkScript)
+	// converting sats/vB -> sats/kvB
+	feeRateAmt := btcutil.Amount(feeRate * 1e3)
+	tx, err := w.wallet.SendOutputs([]*wire.TxOut{wireOP}, nil, w.acctNum, 0, feeRateAmt, "")
+	if err != nil {
+		return nil, err
+	}
+
+	txHash := tx.TxHash()
+
+	return &txHash, nil
+}
+
+func (w *spvWallet) sendWithSubtract(pkScript []byte, value, feeRate uint64) (*chainhash.Hash, error) {
+	const unfundedTxSize = dexbtc.MinimumTxOverhead + 2*dexbtc.P2WPKHOutputSize
+
+	unspents, err := w.listUnspent()
+	if err != nil {
+		return nil, fmt.Errorf("error listing unspent outputs: %w", err)
+	}
+
+	utxos, _, _, err := convertUnspent(0, unspents, w.chainParams)
+	if err != nil {
+		return nil, fmt.Errorf("error converting unspent outputs: %w", err)
+	}
+
+	enough := func(inputsSize, inputsVal uint64) bool {
+		return inputsVal >= value
```

Comment:
Is `inputSize` not needed because this is a subtract-from-amount send (withdraw), where the output amount per se is not the target (fees may eat all that's needed as long as the right amount is consumed by the inputs)?  If that reasoning is correct, let's comment and/or `_` out the first arg to `enough`.